scripts/lsp_types.lua

```diff
@@ -0,0 +1,166 @@
+--[[
+Generates lua-ls annotations for lsp
+USAGE:
+nvim -V1 -es --clean +"lua require('scripts.lsp_types').gen('./lsp.json', 'runtime/lua/vim/lsp/generated_types.lua')"
```

Comment:
Maybe change this to vim/lsp/types/protocol.lua

And add a bug comment on top that this is autogenerated with the command to regenerate them

Reply:
And you can use `nvim -l require...` now.

Reply:
I did test it and it worked